lib/pbkit/pbkit.c

```diff
@@ -2287,183 +2287,126 @@ void pb_end(DWORD *pEnd)
             }
         }
     }
+    return NULL;
 }
 
-
-
-void pb_push1to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1)
+inline void pb_pushto(DWORD subchannel, DWORD *p, DWORD command, DWORD nparam)
 {
 #ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push1to: new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push1to: missing pb_begin earlier\n");
-    pb_PushIndex+=2;
-    pb_PushNext+=2;
-    if (pb_PushIndex>128) debugPrint("pb_push1to: begin-end block musn't exceed 128 dwords\n");
+    if (p!=pb_PushNext) debugPrint("pb_pushto: new write address invalid or not following previous write addresses\n");
+    if (pb_BeginEndPair==0) debugPrint("pb_pushto: missing pb_begin earlier\n");
+    pb_PushIndex += 1 + nparam;
+    pb_PushNext += 1 + nparam;
+    if (pb_PushIndex>128) debugPrint("pb_pushto: begin-end block musn't exceed 128 dwords\n");
 #endif
 
-    *(p+0)=EncodeMethod(subchannel,command,1);
-    *(p+1)=param1;
+    *(p+0)=EncodeMethod(subchannel,command,nparam);
 }
 
-void pb_push2to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2)
+inline DWORD *pb_push1to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1)
 {
-#ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push2to : new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push2to : missing pb_begin earlier\n");
-    pb_PushIndex+=3;
-    pb_PushNext+=3;
-    if (pb_PushIndex>128) debugPrint("pb_push2to: begin-end block musn't exceed 128 dwords\n");
-#endif
+    pb_pushto(subchannel,p,command,1);
+    *(p+1)=param1;
+    return p+2;
+}
 
-    *(p+0)=EncodeMethod(subchannel,command,2);
+inline DWORD *pb_push2to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2)
+{
+    pb_pushto(subchannel,p,command,2);
     *(p+1)=param1;
     *(p+2)=param2;
+    return p+3;
 }
 
-void pb_push3to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2, DWORD param3)
+inline DWORD *pb_push3to(DWORD subchannel, DWORD *p, DWORD command, DWORD param1, DWORD param2, DWORD param3)
 {
-#ifdef DBG
-    if (p!=pb_PushNext) debugPrint("pb_push3to : new write address invalid or not following previous write addresses\n");
-    if (pb_BeginEndPair==0) debugPrint("pb_push3to : missing pb_begin earlier\n");
-    pb_PushIndex+=4;
-    pb_PushNext+=4;
-    if (pb_PushIndex>128) debugPrint("pb_push3to: begin-end block musn't exceed 128 dwords\n");
-#endif
-
-    *(p+0)=EncodeMethod(subchannel,command,3);
+    pb_pushto(subchannel,p,command,3);
     *(p+1)=param1;
     *(p+2)=param2;
     *(p+3)=param3;
 }
```

Comment:
Return value missing?

*(Review other instances, too)*

Reply:
Should be fixed now.